crates/containerd-shim-wasmedge/src/instance.rs

```diff
@@ -523,24 +441,68 @@ mod wasitest {
     }
 }
 
-impl EngineGetter for Wasi {
-    type E = Vm;
-    fn new_engine() -> Result<Vm, Error> {
-        PluginManager::load(None).unwrap();
-        let mut host_options = HostRegistrationConfigOptions::default();
-        host_options = host_options.wasi(true);
-        #[cfg(all(target_os = "linux", feature = "wasi_nn", target_arch = "x86_64"))]
-        {
-            host_options = host_options.wasi_nn(true);
-        }
-        let config = ConfigBuilder::new(CommonConfigOptions::default())
-            .with_host_registration_config(host_options)
-            .build()
-            .map_err(anyhow::Error::msg)?;
-        let vm = VmBuilder::new()
-            .with_config(config)
-            .build()
-            .map_err(anyhow::Error::msg)?;
-        Ok(vm)
+#[cfg(test)]
+mod tests {
+    use std::fs::File;
+
+    use tempfile::tempdir;
+
+    use super::*;
+
+    #[test]
+    fn test_maybe_open_stdio() -> Result<(), Error> {
+        let f = maybe_open_stdio("")?;
+        assert!(f.is_none());
+
+        let f = maybe_open_stdio("/some/nonexistent/path")?;
```

Comment:
👍

Reply:
This is a personal preference. I moved the test down because I felt unit tests should be placed at the bottom of the file. I tried to search for style guide on this but could not find anything. With that said, I haven o strong opinions on where the tests go and I am happy to revert this change.

Reply:
Nevermind, I misunderstood your comment. Yes I will move this to the library crate.